### ISSUE TITLE
(Title edited by Sinbad) Aftermath of leaving Red.

### ISSUE THREAD

**Issue by EpicLPer** [mikeshardmind/SinbadCogs#253]
The cogs you made for Red are really useful and many people still love to use them.

Sending a DM message upon loading each cog just to inform them about some personal drama that went on only shines a bad light on you, the developer, himself but not the whole bot project.

RedBot and the community so far has been really nice and helpful for me and many, MANY others. Just cause something personal between you and the developers of said bot happened, which sometimes yes they can be a bit rude but counting in that they have a MASSIVE userbase to handle this is to be expected, shouldn't be put upon the shoulders of others too.

Please re-consider your choice of handling the situation like this, I and many others in the community who use and sometimes even depend on your cogs would really appreciate it. Thanks!

**Comment by cry8wolf9** [comment]
Is getting a message about a change in the support and why really that bad of a thing? I'm personally fine with it.

**Comment by mikeshardmind** [comment]
People should make informed decisions. I was ousted for reasons many people should take issue with and will not be speaking in detail about this until resolved via other means. Users shohld be aware that what they are reliant on isnt being supported anymore, and given what happenend, they may make their own informed choices, including not to use the now unsupported code.

**Comment by EpicLPer** [comment]
> Users shohld be aware that what they are reliant on isnt being supported anymore
Yes indeed, but not in the way you chose to by showing people a one-sided view of what happened in a really compressed down form.

**Comment by mikeshardmind** [comment]
I won't be providing more information at this time, but I attest that what I already said is representative of what happened.

**Comment by smlbiobot** [comment]
@mikeshardmind Do you have your own Discord server? I actually found out about this repo because of that announcement. I see that it has some good stuff in it and so would like to see if you have your own community server. Thanks!

**Comment by mikeshardmind** [comment]
@smlbiobot I do, but am not currently inviting people to it due to not wanting to deal with people viewing it as a place to discuss the current events which prompted this issue. I may reopen it after this reaches a conclusion of some sort or at least when I stop getting harassing and threatening DMs/emails from people about this.

**Comment by smlbiobot** [comment]
Noted. I will come back and check later (hopefully you will let me know where I can find this link)

**Comment by JordanPlayz158** [comment]
> > Users shohld be aware that what they are reliant on isnt being supported anymore
> > Yes indeed, but not in the way you chose to by showing people a one-sided view of what happened in a really compressed down form.

The question you need to truly ask is what does he have to gain from lying?

**Comment by mikeshardmind** [comment]
I now have the approval to share screenshots that show the full conversational context.I definitely made some mistakes here too, but I was at my wit's end dealing with the way things were handled. There are a few conversations I was advised against sharing, and even if this was not the case, I won't be sifting through everything, this was what happened after a long period.

Since people feel it was too one-sided, this is what I can share at this time. There's, unfortunately, a lot more that can be shared than this, but there's a limit to how far back I'm willing to go sift through conversations for things I can reasonably include the full conversational context to remain within what I was advised was allowed as a defense publicly.

The screenshots are unaltered and span the entire conversation as it happened in each DM. I'm not proud of how I've handled this either, but I'm putting this out here after being told it's okay because I'm tired. I'm very much partially at fault here as well, but I'm tired of accusations of lying, of harassment, and even death threats by email over this. 

This is 26 going into my DMs after https://discordapp.com/channels/133049272517001216/133081046869737472/739508031393497178

![Screenshot_20200806-043725_Discord](https://user-images.githubusercontent.com/22035876/90913807-c6f34a80-e3aa-11ea-9226-636308e869d7.jpg)


Because people may question the edit, the edit was for a typo and was made before the response, I've reopened on desktop to confirm this 

![Discord_WwFGdNwke1](https://user-images.githubusercontent.com/22035876/90913528-63691d00-e3aa-11ea-9c11-da82b3cf18a4.png)

Here's a prior conversation (there's another before this) which was after I retired as a core developer over many of these issues which indicates that 26 was very much previously aware there were issues.

![Discord_pPA6C6HMGE](https://user-images.githubusercontent.com/22035876/90913656-8e537100-e3aa-11ea-8f9a-a0cbe999736e.png)

And here's a conversation between me and Draper which indicates that it wasn't just me perceiving things as issues and that I was not even the only one considering removing contributions based on the behavior of various staff members and them not being accountable to their actions.

![Screenshot_20200821-121917_Discord](https://user-images.githubusercontent.com/22035876/90913889-e68a7300-e3aa-11ea-94aa-fbd446f89953.jpg)

This has been an ongoing issue since prior to Red version 3.2 My contributions wrapping up 3.2 and anything I bothered to assist with after that were with the goal of supporting two communities which I am involved with who were reliant on Red, not in ignoring the issues

Anything in this which is not strictly a screenshot or otherwise asserted as fact is true to the best of my recollection, but I only have precise records of DMs and anything I had the foresight to capture before being banned.

**Comment by mikeshardmind** [comment]
There was a screenshot from draper after "Here's the pins" which he deleted. I can't prove that, but this is the best of what I have available to get people to lay off with thinking I was just doing this in reaction to a ban or lying to hurt the project.

**Comment by mikeshardmind** [comment]
These are two additional screenshots timed with my original resignation as core developer that indicates that other org members were also aware of the prior issues, burnout and that things were not actually being handled.

![Discord_QA2jViStUl](https://user-images.githubusercontent.com/22035876/90917718-6fa4a880-e3b1-11ea-8b3a-c514fa55775b.png)


![image](https://user-images.githubusercontent.com/22035876/90917923-c8744100-e3b1-11ea-9477-dd708306580a.png)

**Comment by mikeshardmind** [comment]
And here's my help being solicited even at a time where I had left the server and been vocal about it, and me still for some reason thinking it was worth actually responding to.

![Discord_u90FACN3JA](https://user-images.githubusercontent.com/22035876/90918203-3f113e80-e3b2-11ea-9e6c-082e39101829.png)

**Comment by mikeshardmind** [comment]
![image](https://user-images.githubusercontent.com/22035876/90922484-112ff800-e3ba-11ea-86bc-09991b4449a5.png)

And this was the announcement which was made. Notice an everyone ping was used as well as referring to a specific commit as "safe" implying that sending a message informing I left was unsafe. Go check where everyone pings have been used in the red server if you feel that's an exaggeration or reading too much into a specific intent here.

**Comment by smooreace** [comment]
Who cares!

**Comment by DiscordLiz** [comment]
> Who cares!

People have been harassing a friend of mine for being willing to actually say something. People accused him of lying, of just trying to tear down the project. He tried to improve things from within the project, left, returned to fix some things, and has now been "ousted" again.

I wish he hadn't cared as much as he did. Red and the people harassing him on their behalf seeing only Red's side of things shouldn't be asking anything of him, especially not for him to just shut up and provide tools they want.

**Comment by mikeshardmind** [comment]
I really hope nobody read what is here as an invitation to harass them in return. Vandalizing their issue tracker does not improve the situation and I do not approve of it. What happened was put out there for people to make their own informed decisions, not to use as fuel for harassment.

**Comment by yenatofu** [comment]
I love the arbitrary excuse that a message on cog load is annoying to the final user.

1: Unlike with the messages that occur when adding repos, it's DM'd to the bot owner,
2: Literally NO ONE is going to load and unload cogs more frequently than monthly (unless editing the code, and if someone's editing the code, they can easily remove the message).

So their childish attempt to dismissively sweep this under the carpet instead of actively promoting discussion (or even engaging in public relations spin) - really breaks my trust in the product and their status as the developers of a bot that represents me on over 1000 servers.

The issue is that there is no consequence of users migrating to a different service, because there is no other choice. So they will continue to act poorly.

And in case the current devs read this: a message on cog load is nowhere near as annoying as being unable to remove the completely off-topic, factually incorrect, spelling-error-laden trivia lists.

**Comment by mon5termatt** [comment]
Just saying, when you try to update it doesnt tell you its from this repo, took me a bit to come say thanks for all you have done!

good luck in your future endeavors :)

**Comment by rmscode** [comment]
Hey man . . . Your unfortunate drama with Red aside, how do I contact you outside of github?

**Comment by mikeshardmind** [comment]
@RS7588 I don't mean any offense by this, but I'm not exactly eager to have people reaching out relating to this repo after all of what has happened. There's an email in the code of conduct file. if you remove the `-conduct` portion, you'll at least have the correct point of contact. I'm not going to guarantee a response to anything related to this repo right now. If you are reaching out for things not related to this repo, please contact from a more relevant location as I'm much more likely to take the time on it.

**Comment by mikeshardmind** [comment]
After some time more distanced from this and time taken to balance the various values at play, there's an update that makes an attempt to only send it once per bot start with more clarity as to where the message came from. If you're restarting your bot so frequently that this is an issue, fix the thing causing that to be needed.

**Comment by EpicLPer** [comment]
> After some time more distanced from this and time taken to balance the various values at play, there's an update that makes an attempt to only send it once per bot start with more clarity as to where the message came from. If you're restarting your bot so frequently that this is an issue, fix the thing causing that to be needed.

Still not a good solution...

**Comment by mikeshardmind** [comment]
> Still not a good solution...

Well, you're free to not use the code here then.

I've been considering just swearing off open-source altogether after all of this. Why invest my time? It's basically a job where you get harassed by people who don't have the context you did when making decisions, no appreciable amount of pay for the majority of developers. Bringing up issues with code gets taken personally by people rather than as a cause to fix the code, bringing up issues with policies or the need to have more activity gets treated similarly. Then if you dare speak up about the experience afterward with even a fraction of the visibility of the original project, you get told you shouldn't have said anything.

That's the reality of it. If you don't like that, go pay for someone to make something that works exactly how you want or go make it yourself.

**Comment by Zephyrkul** [comment]
Thank you.

**Comment by crunchprank** [comment]
Hi Mike,

I've noticed some activity in your repo lately. I was curious to know if you have decided to continue development (for the time being)? Additionally, I'd like to know if you had a Discord server? I understand there's quite a bit of theatrics currently, but I'm simply looking for a resource to have for communication purposes and/or support if needed - not to engage in anything regarding the circumstances of the current situation.

Thanks!

**Comment by EpicLPer** [comment]
> Hi Mike,
> 
> I've noticed some activity in your repo lately. I was curious to know if you have decided to continue development (for the time being)? Additionally, I'd like to know if you had a Discord server? I understand there's quite a bit of theatrics currently, but I'm simply looking for a resource to have for communication purposes and/or support if needed - not to engage in anything regarding the circumstances of the current situation.
> 
> Thanks!

People were editing his cogs so they wouldn't DM you upon loading them anymore. He now, as it seems by the last commits, implemented a way to not circumvent this anymore.  
In short he made it even worse than it was before which is a real d\*\*\* move...

**Comment by mikeshardmind** [comment]
@crunchprank recent activity was related to honoring a clause of relicensing (see #258). As a result of this, the history of the repo ended up altered, and certain things like feedreaders may register those commits as new activity.

**Comment by mikeshardmind** [comment]
> People were editing his cogs so they wouldn't DM you upon loading them anymore. He now, as it seems by the last commits, implemented a way to not circumvent this anymore.
In short he made it even worse than it was before which is a real d*** move.

@EpicLPer 

The only thing that's done as "anti-circumvention" prevents *Red* from circumventing it automatically as I was sent screenshots of people suggesting that Red do this. Considering the actions of the remaining organization members are directly responsible for my decision to send that message, I took issue with the idea of them circumventing it. If you see a developer informing people that something isn't supported anymore and why as a bad thing, feel free not to use the no longer supported code.

If you want to continue this with some amount of civility, that's fine, but consider this a final warning for what I will tolerate.

**Comment by EpicLPer** [comment]
Ah yes lol. Nice.

**Comment by luna-system** [comment]
As a small aside - are there other bot cores like this for Discord? ones where the core functionality is minimal and meant to be extended by users (as opposed to other big bots who tend to do a little bit of everything and don't have much extendability)?

**Comment by EpicLPer** [comment]
> As a small aside - are there other bot cores like this for Discord? ones where the core functionality is minimal and meant to be extended by users (as opposed to other big bots who tend to do a little bit of everything and don't have much extendability)?

Use Red Bot. Just cause Sinbad seems to not accept that the Red Bot team can’t always comply with other people’s wishes doesn’t mean the project as a whole is bad.
So far he seems to be one of few that complain about stuff, and judging by the way he handled the situation here it was bound to explode at some point...

Red Bot is a good project, some people don’t seem to get that sadly.

**Comment by luna-system** [comment]
> > As a small aside - are there other bot cores like this for Discord? ones where the core functionality is minimal and meant to be extended by users (as opposed to other big bots who tend to do a little bit of everything and don't have much extendability)?
> 
> Use Red Bot. Just cause Sinbad seems to not accept that the Red Bot team can’t always comply with other people’s wishes doesn’t mean the project as a whole is bad.
> So far he seems to be one of few that complain about stuff, and judging by the way he handled the situation here it was bound to explode at some point...
> 
> Red Bot is a good project, some people don’t seem to get that sadly.

With all due respect, this is not an answer to my question, nor the potential question others may ask themselves if/when they stumble on this the same way I did.

**Comment by rmscode** [comment]
> > > As a small aside - are there other bot cores like this for Discord? ones where the core functionality is minimal and meant to be extended by users (as opposed to other big bots who tend to do a little bit of everything and don't have much extendability)?
> > 
> > 
> > Use Red Bot. Just cause Sinbad seems to not accept that the Red Bot team can’t always comply with other people’s wishes doesn’t mean the project as a whole is bad.
> > So far he seems to be one of few that complain about stuff, and judging by the way he handled the situation here it was bound to explode at some point...
> > Red Bot is a good project, some people don’t seem to get that sadly.
> 
> With all due respect, this is not an answer to my question, nor the potential question others may ask themselves if/when they stumble on this the same way I did.

You ask a great question. Something I've wondered myself as well. I'm just curious to know if there is anything else out there that is comparable. I always like to play with new things. (Sorry that this isn't an answer)

**Comment by mikeshardmind** [comment]
> As a small aside - are there other bot cores like this for Discord? ones where the core functionality is minimal and meant to be extended by users (as opposed to other big bots who tend to do a little bit of everything and don't have much extendability)?

I'm not really aware of others specifically intended to be extended by other users in a mature state (Not that I've been exactly looking since I'm more just handling what I want from here), but any discord.py-based bot can be extended by creating extensions, this isn't an invention of Red. ([Docs](https://discordpy.readthedocs.io/en/latest/#extensions)), and there are examples of extensions intended to be loadable by nearly any discord.py bot like [Jishaku](https://github.com/Gorialis/jishaku)

**Comment by luna-system** [comment]
> > As a small aside - are there other bot cores like this for Discord? ones where the core functionality is minimal and meant to be extended by users (as opposed to other big bots who tend to do a little bit of everything and don't have much extendability)?
> 
> I'm not really aware of others specifically intended to be extended by other users in a mature state (Not that I've been exactly looking since I'm more just handling what I want from here), but any discord.py-based bot can be extended by creating extensions, this isn't an invention of Red. ([Docs](https://discordpy.readthedocs.io/en/latest/#extensions)), and there are examples of extensions intended to be loadable by nearly any discord.py bot like [Jishaku](https://github.com/Gorialis/jishaku)

Sure, I just mean other projects that are in the Hubot style of "here's a basic self-hosted bot that exists, and a folder to drop your custom extensions into" sort of deal

**Comment by mikeshardmind** [comment]
I'll have some news on a minimal bot core that can be extended over [here](https://github.com/unified-moderation-network/salamander). I would not consider it to be in a state where it qualifies as a stable core at the moment, but given what I've observed with continuing to not handle things appropriately from Red's devs (see this [additional needed fix](https://github.com/Cog-Creators/Red-DiscordBot/pull/4556) to something they considered enough of a security issue to publish an [advisory](https://github.com/Cog-Creators/Red-DiscordBot/security/advisories/GHSA-mp9m-g7qj-6vqr) for in which the original fix was obviously not tested because it would have errored out) I see a greater need to answer the question for those actually looking for something of this nature.

This isn't the first time they've handled high impact issues in a laqsidasical manner, and that's clearly not changing.

Edit: And [another issue with the same fix](https://github.com/Cog-Creators/Red-DiscordBot/pull/4575) that would have been noticed had anyone tested it

**Comment by TeemoTheYiffer** [comment]
Hey Sinbad,

Long time no see. How's my favorite Python Jesus doing? Keeping that handsome lush hair well maintained I hope.

I noticed you go dark several months ago. I got banned but it was reversed and went to look for you. Everyone was hush-hush-see-pinned about your disappearance. I'm glad I found this to find the whole story. I warned you about them, didn't I? It was a year ago I believe. There is a reason why I had quarter of the staff blocked. I knew you were doing a lot, judging by PRs & code reviews. Didn't know it was this much. 

Well if you ever want to rekindle, you know where to reach me.

Take care of yourself, especially in that hot & stinky state of yours~

-Teemo the Yiffer
